tests/e2e/uninstall.go

```diff
@@ -0,0 +1,194 @@
+// Copyright © 2023 Cisco Systems, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"fmt"
+
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+// requireRemoveKoperatorCRDs deletes the koperator CRDs
+func requireRemoveKoperatorCRDs(kubectlOptions k8s.KubectlOptions) {
+	It("Removing koperator CRDs", func() {
+		for _, crd := range koperatorCRDs() {
+			deleteK8sResourceNoErrNotFound(kubectlOptions, defaultDeletionTimeout, crdKind, crd)
+		}
+	})
+}
+
+// requireUninstallingKoperator uninstall koperator Helm chart and removes Koperator's CRDs.
+func requireUninstallingKoperator(kubectlOptions k8s.KubectlOptions) {
+	When("Uninstalling Koperator", func() {
+		requireUninstallingKoperatorHelmChart(kubectlOptions)
+		requireRemoveKoperatorCRDs(kubectlOptions)
+	})
+}
+
+// requireUninstallingKoperatorHelmChart uninstall Koperator Helm chart
+// and checks the success of that operation.
+func requireUninstallingKoperatorHelmChart(kubectlOptions k8s.KubectlOptions) {
+	It("Uninstalling Koperator Helm chart", func() {
+		uninstallHelmChartIfExists(kubectlOptions, koperatorLocalHelmDescriptor.ReleaseName, true)
+		By("Verifying Koperator helm chart resources cleanup")
+		k8sResourceKinds, err := listK8sAllResourceKind(kubectlOptions)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		koperatorAvailableResourceKinds := stringSlicesUnion(koperatorCRDs(), k8sResourceKinds)
+		koperatorAvailableResourceKinds = append(koperatorAvailableResourceKinds, basicK8sResourceKinds()...)
+
+		remainedResources, err := getK8sResources(kubectlOptions,
+			koperatorAvailableResourceKinds,
+			fmt.Sprintf(managedByHelmLabelTemplate, koperatorLocalHelmDescriptor.ReleaseName),
+			"",
+			kubectlArgGoTemplateKindNameNamespace,
+			"--all-namespaces")
+
+		Expect(err).ShouldNot(HaveOccurred())
+		Expect(remainedResources).Should(BeEmpty())
+	})
```

Comment:
**Note**: This is fine like this for now, but in a follow-up ticket we want to move these kind of checks higher up onto case level when this becomes unnecessary.
Then everything will fold up because the Helm chart uninstallation becomes only 1 line, the Koperator uninstallation becomes only 1 line so we can omit those and put the uninstall Koperator `It(...)` directly into the uninstall test case.

Same thing I did at the last refactor of the install test case.

Same holds true for the other components.

CC: @panyuenlau @Kuvesz @mihaialexandrescu

Reply:
I dont understand what do you mean.

Reply:
I know you put in your test_install.go directly those tests under When containers. Somehow I felt that it is clener and maybe more reusable if I kept the original form and put evreything in functions.

Reply:
I wrote in private.